apps/projects/routes.py

```diff
@@ -752,22 +671,218 @@ def delete(self, project_id):
+                      width:
+                        type: int
+                        example: 640
+                      height:
+                        type: int
+                        example: 360
+                      duration:
+                        type: float
+                        example: 300.014000
+                      bit_rate:
+                        type: int
+                        example: 287654
+                      nb_frames:
+                        type: int
+                        example: 7654
+                      r_frame_rate:
+                        type: string
+                        example: 24/1
+                      format_name:
+                        type: string
+                        example: mov,mp4,m4a,3gp,3g2,mj2
+                      size:
+                        type: int
+                        example: 14567890
+                  url:
+                    type: string
+                    example: http://localhost:5050/projects/5cbd5acfe24f6045607e51aa/raw/video
+                  mime_type:
+                    type: string
+                    example: video/mp4
+                  create_time:
+                    type: string
+                    example: 2019-07-02T15:02:32+00:00
+                  original_filename:
+                    type: string
+                    example: video.mp4
+                  request_address:
+                    type: string
+                    example: 127.0.0.1
+                  version:
+                    type: integer
+                    example: 1
+                  parent:
+                    type: string
+                    example: 5d2c8078fe985e7587b50de7
+                  processing:
+                    type: object
+                    properties:
+                      video:
+                        type: boolean
+                        example: False
+                      thumbnail_preview:
+                        type: boolean
+                        example: False
+                      thumbnails_timeline:
+                        type: boolean
+                        example: False
+                  thumbnails:
+                    type: object
+                    properties:
+                      timeline:
+                        type: array
+                        example: []
+                      preview:
+                        type: object
+                        example: {}
+        """
+
+        if any(self._project['processing'].values()):
+            return json_response({"processing": True}, status=202)
+
+        # deepcopy & save a child_project
+        child_project = copy.deepcopy(self._project)
+        del child_project['_id']
+        del child_project['storage_id']
+        child_project['parent'] = self._project['_id']
+        child_project['create_time'] = datetime.utcnow()
+        child_project['version'] += 1
```

Comment:
Would it reasonable, we increase the version when we `duplicate`? 
When we `edit` original video, we `duplicate` and `PUT edit` video, that mean video jump from 1 to 3, it have little confusion and `duplicate` is only copy video haven't anything changes.
So I think we shouldn't increase version when `duplicate`.
and https://github.com/superdesk/video-server/pull/11/files#r305230095
To check only `PUT edit` duplication, we use `parent_id` instead for `version`.

Reply:
I increment a version during duplicate, because a Superdesk has the same logic for this case.
I think it's not a big problem that we have a version 3 after first edit.
original project - version 1
duplicated project - version 2
edited project - version 3

Let's ask for one more pair if eyes :)  @petrjasek

Reply:
I think it's ok as long is it's always going up so you can compare which one is newer